cdap-api/src/main/java/co/cask/cdap/internal/schedule/constraint/Constraint.java

```diff
@@ -19,5 +19,5 @@
 /**
  * A constraint that is checked before executing a schedule.
  */
-public abstract class Constraint {
+public interface Constraint {
```

Comment:
@chtyim suggested that the class in cdap-api should remain an abstract class instead of an interface.
Though, I do agree that it is simpler to implement the class hierarchy of constraints if we have it be an Interface.